test/index.js

```diff
@@ -0,0 +1,988 @@
+/**
+ * @description   Tests for signalk-js-sdk. Also useful for spec-testing a SK server.
+ * @author        Fabian Tollenaar <fabian@decipher.industries>
+ * @copyright     2018-2019, Fabian Tollenaar. All rights reserved.
+ * @license       Apache-2.0
+ * @module        @signalk/signalk-js-sdk
+ */
+
+import mdns from 'mdns'
+import Client, { Discovery, Client as NamedClient, PERMISSIONS_READONLY } from '../src'
+import { assert } from 'chai'
+import { v4 as uuid } from 'uuid'
+
+const isObject = (mixed, prop, propIsObject) => {
+  const _isObj = mixed && typeof mixed === 'object'
+
+  if (!_isObj) {
+    return false
+  }
+
+  if (typeof prop === 'string' && typeof propIsObject === 'boolean') {
+    const _propIsObj = mixed[prop] && typeof mixed[prop] === 'object'
+    return _isObj && mixed.hasOwnProperty(prop) && _propIsObj === propIsObject
+  }
+
+  if (typeof prop === 'string') {
+    return _isObj && mixed.hasOwnProperty(prop)
+  }
+
+  return _isObj
+}
+
+const TEST_SERVER_HOSTNAME = (isObject(process, 'env', true) && isObject(process.env, 'TEST_SERVER_HOSTNAME')) ? process.env.TEST_SERVER_HOSTNAME : 'hq.decipher.digital'
+const TEST_SERVER_PORT = (isObject(process, 'env', true) && isObject(process.env, 'TEST_SERVER_PORT')) ? process.env.TEST_SERVER_PORT : 3000
+
+describe('Signal K SDK', () => {
+  // @TODO requesting access should be expanded into a small class to manage the entire flow (including polling)
+  describe('Device access requests', () => {
+    it('... successfully requests device access', done => {
```

Comment:
very minor style issue: mocha output already shows a nice itemized list, why the three dots?